smartctl.go

```diff
@@ -42,6 +43,16 @@ type SMARTctl struct {
 	device SMARTDevice
 }
 
+func extractDiskName(input string) string {
+	re := regexp.MustCompile(`^(?:/dev/\S+/\S+\s\[|/dev/|\[)(?:\s\[|)(?P<disk>[a-z0-9_]+)(?:\].*|)$`)
+	match := re.FindStringSubmatch(input)
+
+	if len(match) > 0 {
+		return match[re.SubexpIndex("disk")]
+	}
+	return ""
```

Comment:
this `return ""` feels weird; falling back to a non-zero string is probably safer.

Reply:
maybe but I didn't change logic of function,  I only change method to extract string.

Reply:
The `info_name` part is key here, I think we just need a large corpus to verify.